src/Service/Models/GraphQLFilterParsers.cs

```diff
@@ -113,13 +114,60 @@ public GQLFilterParser(ISqlMetadataProvider metadataProvider)
                     List<ObjectFieldNode> subfields = (List<ObjectFieldNode>)fieldValue;
 
                     // Preserve the name value present in the filter.
+                    // In some cases, 'name' represents a relationship field on a source entity,
+                    // and not a field on a relationship target entity.
+                    // Additionally, 'name' may not be the same as the relationship's target entity because
+                    // runtime configuration supports overriding the entity name in the source entity configuration.
+                    // e.g.
+                    // comics (Source entity)  series (Target entity)
+                    // - id                    - id
+                    // - myseries [series]     - name
+                    // e.g. GraphQL request
+                    // {  comics ( filter: { myseries: { name: { eq: 'myName' } } } ) { <field selection> } }
                     string backingColumnName = name;
+
                     _metadataProvider.TryGetBackingColumn(queryStructure.EntityName, field: name, out string? resolvedBackingColumnName);
+
+                    // When runtime configuration defines relationship metadata,
+                    // an additional field on the entity representing GraphQL type
+                    // will exist. That relationship field does not represent a column in
+                    // the representative database table and the relationship field will not have an authorization
+                    // rule entry. Authorization is handled by permissions defined for the relationship's
+                    // target entity.
+                    bool relationshipField = true;
                     if (!string.IsNullOrWhiteSpace(resolvedBackingColumnName))
                     {
                         backingColumnName = resolvedBackingColumnName;
+                        relationshipField = false;
+                    }
+
+                    // Only perform field (column) authorization when the field is not a relationship field.
+                    // Due to the recursive behavior of SqlExistsQueryStructure compilation, the column authorization
+                    // check only occurs when access to the column's owner entity is confirmed.
+                    if (!relationshipField)
+                    {
+                        // GetTarget will return the name of the target entity when the filterArgumentSchema has a relationship directive.
+                        // GetTarget returns null when the filterArgumentSchema does not have a relationship directive.
+                        // No relationship directive indicates that the input field does not represent a nested filter.
+                        string targetEntity = RelationshipDirectiveType.GetTargetEntity(filterArgumentSchema) ?? queryStructure.EntityName;
```

Comment:
If its not a relationshipField, why do we need to GetTargetEntity from the relationship directive?

Reply:
Using this nested filter example:
```graphql
            series ( filter: { comics: { categoryName: { contains: ""Tales"" }}} )
                    {
                      items {
                        id
                        name
                      }
                    }
```
when processing "comics" in the filter, comics is an InputObject named ComicFilterInput and is an input field of "series". Because ComicFilterInput is used to traverse the relationship between series and comic, ComicFilterInput has a relationship directive whose target entity name is Comic. That value (target entity name) is needed to perform authz permissions lookups, as we don't have a way to identify the target entity otherwise.

Now in the case of a non-nested filter input, on an entity with no relationships defined, there is no opportunity to identify the target entity name from a relationship directive.
```graphql
                websiteUsers( filter: : {username: {isNull: false}}) {
                    items {
                        id
                        username
                    }
                }
```
In this case, websiteUserFilterInput is passed to RelationshipDirectiveType.GetTargetEntity, which returns null, so the entity name (WebsiteUser) is resolved from `queryStructure.EntityName`

This code path covers both nested and non-nested filter processing, by enabling the identification of the entityname the filter belongs to.

Reply:
I just removed  `RelationshipDirectiveType.GetTargetEntity(filterArgumentSchema)` and tests still pass. I had encountered situations where queryStructure.EntityName seemed to not give me the correct entity, but that may have been when i was still in progress on this change and is no longer an issue.

Reply:
I thought the reason why RelationshipDirectiveType.GetTargetEntity is not needed here is because its gated under the check `if (!relationshipField)` - I wonder what codepath is picked up when its a relationship field

Reply:
`if (!relationshipField)` only gates the column evaluation. When the field is a relationship field, the only difference is that column evaluation in authorizationResolver is not needed.

Reply:
why is the column evaluation not needed? Do we disallow relationship fields from being present in the policy clause - i.e. validate only the columns of the source entity are referenced in the policy?

Probably, providing support for adding relationship field in the policy is something complex policies support would provide.

Reply:
A relationship field doesn't represent a field/column in the database and acts as a mechanism for traversing relationships in the GraphQL input filter type.

When referencing those relationship fields outside of a filter e.g. graphql request body, then the HotChocolate authz system performs field authorize as directed by `@authorize` directives on the schema.

Reply:
I do agree that we can possibly utilize the relationship field for complex policies.

Reply:
fyi: @ayush3797 potential to use relationship fields as a mechanism to navigate to nested entities for complex policies

Reply:
To answer your question: 
> "Do we disallow relationship fields from being present in the policy clause - i.e. validate only the columns of the source entity are referenced in the policy?"
We disallow relationship fields from being present in the policy clause. This is facilitate by the OData filter parser referencing the EdmModel that is created during startup. When database policies are processed on the SqlQueryStructure, there would be an OData error raised when `GetODataParser().GetFilterClause()` is called from `AuthorizationPolicyHelpers`